src/Tools/dotnet-stack/Symbolicate.cs

```diff
@@ -0,0 +1,426 @@
+                StreamWriter fswo = null;
+                if (outputPath != null)
+                {
+                    fswo = new StreamWriter(new FileStream(outputPath, FileMode.Create, FileAccess.Write));
+                    output = $"\nOutput: {outputPath}\n";
+                }
+
+                using StreamReader fsri = new StreamReader(new FileStream(inputPath, FileMode.Open, FileAccess.Read));
+                while (!fsri.EndOfStream)
+                {
+                    string line = fsri.ReadLine();
+                    // The stacktrace must have "at ... in ... token ..."
+                    if (!line.Contains("at ") || !line.Contains(" in ") || !line.Contains("token"))
+                    {
+                        fswo?.WriteLine(line);
+                        console.Out.WriteLine($"{line}");
+                        continue;
+                    }
+                    string ret = GetRegex(console, line, xmlList, fswi);
+                    fswo?.WriteLine(ret);
+                    console.Out.WriteLine($"{ret}");
+                }
+                fswo?.Close();
+                fswi?.Close();
+                console.Out.WriteLine($"{output}");
+            }
+            catch (Exception e)
+            {
+                console.Error.WriteLine(e.Message);
+            }
+        }
+
+        private static string GetRegex(IConsole console, string line, List<string> xmlList, StreamWriter fsw)
+        {
+            string ret = line;
+            Match match = regex.Match(line);
+            if (!match.Success && fsw != null)
+            {
+                ret = VerifyStackTraceLine(line);
+                if (ret == line)
+                {
+                    fsw.WriteLine(line);
+                    return line;
+                }
+                match = regex.Match(ret);
+            }
+            fsw?.WriteLine(ret);
+
+            StackTraceInfo stInfo = new StackTraceInfo() {
+                Type = match.Groups["type"].Value,
+                Method = match.Groups["method"].Value,
+                Param = match.Groups["params"].Value,
+                Assembly = match.Groups["filename"].Value,
+                Token = match.Groups["token"].Value,
+                Offset = match.Groups["offset"].Value
+            };
+
+            string xmlStr = stInfo.Assembly.Contains(".ni.dll") ? stInfo.Assembly.Replace(".ni.dll", ".xml") : stInfo.Assembly.Replace(".dll", ".xml");
+            foreach (var xmlPath in xmlList)
+            {
+                if (xmlPath.Contains(xmlStr))
+                {
+                    GetLineFromXml(console, xmlPath, stInfo);
+                    if (stInfo.Filepath != null && stInfo.StartLine != null)
+                    {
+                        return $"   at {stInfo.Type}.{stInfo.Method}({stInfo.Param}) in {stInfo.Filepath}:line {stInfo.StartLine}";
+                    }
+                }
+            }
+            return ret;
```

Comment:
This loop is expensive since it will potentially iterate the entire list of XML files for every line that matches the regex. This will most likely go away when you switch to using `System.Reflection.Metadata` for looking up symbols though.